internal/regionsrv/hostsfile_test.go

```diff
@@ -53,6 +53,12 @@ func TestUpdateHostsFileCouldNotRead(t *testing.T) {
 }
 
 func TestUpdateHostsFileCouldNotWrite(t *testing.T) {
+	// TODO: is there a better way to test this?
+	//       root can read/write anything
+	if os.Getuid() == 0 {
```

Comment:
I wonder if we shouldn't simply have a test that fails if `os.Geteuid()` returns 0 rather than hacking on the tests to pass as root. 

I personally would never run a unittest suite as root. I am not sure we should go down the path of supporting that, especially when it means that some of the tests are skipped then. that basically means the testing coverage is not the same. and that's bad.

Reply:
Honestly, in a perfect scenario, the test case should not care about which user is being run. I find it completely wrong to assume in a test that a given path is not readable, it can be readable for certain users.

Some tests IMHO did not make much sense the way they were written. Code coverage is one thing, the usefulness of a test is another.

Many tests have an assumption that things can't be read just because the user has no permission, but container-suseconnect runs as a zypper plugin, which I believe runs as root, and in theory has permissions for everything.

Reply:
This project is rather small. testing error paths is important for customer resiliency. unit tests should be testing error and edge conditions, as they tend to break during refactorings. It'd be fine to use mocking to reliably trigger specific error conditions and test the code that way. without that, documenting the sensible assumption of somebody not running unittest code as root would be imho a better way than skipping tests when you run it anyhow as root. it makes the assumption explicit rather than hiding it under the rug. 

there can be other reasons why a file could not be written to, even as root, so the condition is valid to test I think.